### PR TITLE
Stubs for visiting MIR nodes

### LINKED ISSUE #11: Set up visitor infrastructure for MIR
Write the code to 
a) find every function/method body in the crate being compiled and
b) load MIR for the body and then traverse the body, visiting every MIR instruction for abstract interpretation.

### PR DESCRIPTION
**Pull request by hermanventer**
## Description

Flesh out the visitor class to include stubs for MIR nodes.

Fixes #11

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] API change with a documentation update

## How Has This Been Tested?

tests/run-pass/crate_traversal.rs has been fleshed out with bodies that cause all the new visitor methods to get called.

### CODE REVIEW

File: tests/run-pass/crate_traversal.rs [hunk 1]
```diff
@@ -1,16 +1,164 @@
...
 }
 
 impl TestStruct {
-    fn test_function() {}
-    fn test_method(&self) {}
+    pub fn test_function() {}
+    pub fn test_method(&self) {}
+}
+
+pub unsafe fn test1() {
```
Comment (sblackshear) [inline tests/run-pass/crate_traversal.rs]: Some test method names and/r comments describing which features each test is targeting might be useful? I can figure some of them out, but not all :).

File: tests/run-pass/crate_traversal.rs [hunk 2]
```diff
@@ -1,16 +1,164 @@
...
+}
+
+fn test2() {
+    let z = 0;
+    let _x = {
+        &z
+    };
+}
+
+fn test3() {
```
Comment (sblackshear) [inline tests/run-pass/crate_traversal.rs]: How do these tests work? Is there an expected output file somewhere?

Comment (hermanventer) [reply tests/run-pass/crate_traversal.rs]: At the moment the role of the test file is just to trigger calls to all of the visitor methods. They don't do anything right now, so there is no output.

File: tests/run-pass/crate_traversal.rs (hunk 1 above)
Comment (hermanventer) [reply tests/run-pass/crate_traversal.rs]: In general, yes, such comments are very helpful (and I wish the rust compiler's test suite had more of them). This test file, however, is of a temporary nature and will become redundant as more useful tests with expected behavior and so on come along. See issue #10. Adding comments detailing with MIR features each test uses would be very tedious and is not a top priority for me right now.

**Pull request merged**